File: Tutoriels/Cpp/POO/Polymorphisme/src/GPersonProfessor.cpp
//===============================================
#include "GPersonProfessor.h"
//===============================================
GPersonProfessor::GPersonProfessor() {
    m_lesson = "";
}
//===============================================
GPersonProfessor::GPersonProfessor(const string& name, const string& email, const string& lesson) :
    GPerson(name, email) {
    m_lesson = lesson;
}
//===============================================
GPersonProfessor::~GPersonProfessor() {

}
//===============================================
void GPersonProfessor::print() const {
    string lData = "";
    lData += "Professeur: Je suis Professeur de ";
    lData += m_lesson;
    lData += "\n";
    cout << lData.c_str();
}
//===============================================

File: Tutoriels/Python/POO/Heritage/src/GPersonProfessor.py
#================================================
from GPerson import GPerson
#================================================
class GPersonProfessor(GPerson): 
    def __init__(self, name, lesson):
        GPerson.__init__(self, name)
        self.m_lesson = lesson
    def __str__(self):
        return "Professeur: {0} | {1}".format(GPerson.__str__(self), self.m_lesson)
#================================================
